src/library/scala/collection/Iterable.scala

```diff
@@ -807,6 +807,16 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     */
   def inits: Iterator[C] = iterateUntilEmpty(_.init)
 
+  /** Applies a side-effecting function to each element in this collection. Strict
+    * collections like Seq's will apply `f` to their elements immediately, while
+    * lazy collections like Views will only apply `f` when elements are evaluated.
+    *
+    * @param f the function to apply to each element
+    * @tparam U the return type of f
+    * @return The same logical collection as this
+    */
+  def tapEach[U](f: A => U): C = fromSpecific(new View.TapEach(this, f))
```

Comment:
I'm also not sure if we really need a `View.TapEach`, or can just do `map(a => { f(a); a })` instead of `fromSpecific(new View.TapEach(this, f))`

Reply:
You're right, I'll change this